mlflow/tracing/artifact_utils.py

```diff
@@ -0,0 +1,20 @@
+from mlflow.entities.trace_info import TraceInfo
+from mlflow.exceptions import MlflowException
+from mlflow.utils.mlflow_tags import MLFLOW_ARTIFACT_LOCATION
+
+TRACE_DATA_FILE_NAME = "traces.json"
+
+
+def get_artifact_uri_for_trace(trace_info: TraceInfo) -> str:
+    """
+    Get the artifact uri for accessing the trace data.
+
+    The artifact root is specified in the trace tags, which is
+    set when logging the trace in the backend.
+    """
+    if MLFLOW_ARTIFACT_LOCATION not in trace_info.tags:
+        raise MlflowException(
+            "Trace artifact location not specified, please specify it with "
+            f"tag '{MLFLOW_ARTIFACT_LOCATION}' in the trace."
+        )
```

Comment:
We might want to modify the message. This sounds like `MLFLOW_ARTIFACT_LOCATION` is mutable but it must be immutable to prevent security vulnerabilities.

Reply:
That's true... probably sth like `Unable to determine trace artifact location`? It seems there is no action users can take if encounter this exception.